hello_world/main.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+from typing import Any
+
+import pkg_resources
+from helpers import make_table
+from prompt_toolkit.application import Application
+from prompt_toolkit.application.current import get_app
+from prompt_toolkit.key_binding import KeyBindings
+from prompt_toolkit.key_binding.bindings.focus import focus_next, focus_previous
+from prompt_toolkit.layout import HSplit, Layout, VSplit
+from prompt_toolkit.styles import Style
+from prompt_toolkit.widgets import Box, Button, Label, TextArea
+
+
+# Event handlers for all the buttons.
+def get_pkg_info() -> str:
+    """
+    Function to give the package information in form of  a table.
+
+    params: none
+    return: string of rows and columns
+    """
+    res = []
+    for pkg in pkg_resources.working_set:
+        res.append([pkg.project_name, pkg.parsed_version])
+    labels = ["Package Name", "Version"]
+    return make_table(res, labels=labels)
+
+
+def spit_pkg_details() -> str:
+    """Sets the text in the text area of the layout."""
+    text_area.text = get_pkg_info()
+
+
+def exit_clicked() -> Any:
+    """When the button is clicked, the app will exit following function provides help in doing that."""
+    get_app().exit()
+
+
+# All the widgets for the UI.
+button1 = Button("Venv Details", handler=spit_pkg_details)
+button4 = Button("Exit", handler=exit_clicked)
+text_area = TextArea(focusable=True)
+
+
+# Combine all the widgets in a UI.
+# The `Box` object ensures that padding will be inserted around the containing
+# widget. It adapts automatically, unless an explicit `padding` amount is given.
+root_container = Box(
+    HSplit(
+        [
+            Label(text="Press `Tab` to move or use 'mouse' "),
+            VSplit(
+                [
+                    Box(
+                        body=HSplit([button1, button4], padding=1),
+                        padding=1,
+                        style="class:left-pane",
+                    ),
+                    Box(body=text_area, padding=1, style="class:right-pane"),
+                ]
+            ),
+        ]
+    ),
+)
+
+layout = Layout(container=root_container, focused_element=button1)
+
+
+# Key bindings.
+kb = KeyBindings()
+kb.add("tab")(focus_next)
+kb.add("s-tab")(focus_previous)
+
+
+# Styling.
+style = Style(
+    [
+        ("left-pane", "bg:#888800 #000000"),
+        ("right-pane", "bg:#000000 #ffffff"),
+        ("button", "#000000"),
+        ("button-arrow", "#0000ff"),
+        ("button focused", "bg:#ff0000"),
+        ("text-area focused", "bg:#ff0000"),
+    ]
+)
+
+
+# Build a main application object.
+application = Application(
+    layout=layout, key_bindings=kb, style=style, full_screen=True, mouse_support=True
+)
+
+
+def main() -> Any:
```

Comment:
```suggestion
def main() -> None:
```

It doesn't return anything.

Reply:
:thinking:  isn't `run` a callable?

Reply:
`run` is a callable but that's not what is returned, it is just ran inside the function. By default if you don't return anything from a function, python automatically returns `None` if you don't return anything.

Reply:
I see thanks!